src/action/base/move_unpacked_nix.rs

```diff
@@ -110,13 +109,21 @@ impl Action for MoveUnpackedNix {
                 .map_err(|e| ActionErrorKind::Rename(entry.path(), entry_dest.to_owned(), e))
                 .map_err(Self::error)?;
 
-            let perms: Permissions = PermissionsExt::from_mode(0o555);
             for entry_item in WalkDir::new(&entry_dest)
                 .into_iter()
                 .filter_map(Result::ok)
                 .filter(|e| !e.file_type().is_symlink())
             {
-                tokio::fs::set_permissions(&entry_item.path(), perms.clone())
+                let path = entry_item.path();
+
+                let mut perms = path
+                    .metadata()
+                    .map_err(|e| ActionErrorKind::GetMetadata(path.to_owned(), e))
+                    .map_err(Self::error)?
+                    .permissions();
+                perms.set_readonly(true);
```

Comment:
I believe I would be more comfortable setting a mode like before? I'm not sure if this specifically does what we need? https://doc.rust-lang.org/std/fs/struct.Permissions.html#method.set_readonly

Reply:
The benefit of `set_readonly()` is that we don't need to compute a mode at all: it's the equivalent of `chmod a-w file` without needing to do `chmod (get_current_mode(file) & 0o555) file`.

I'm like 67% sure I tested some code that basically does this over the weekend and it did what I expected (i.e. 777 became 555, 666 became 444).